### PR TITLE
OSOE-867: Addressing analyzer warnings

### PR DESCRIPTION
**Pull request by Psichorex**
[OSOE-867](https://lombiq.atlassian.net/browse/OSOE-867)

[OSOE-867]: https://lombiq.atlassian.net/browse/OSOE-867?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.TrainingDemo/Controllers/ApiController.cs [hunk 1]
```diff
@@ -29,6 +29,10 @@ namespace Lombiq.TrainingDemo.Controllers;
 // endpoints should most of the time use the "Api" authentication scheme: This is not the same that standard users are
 // authenticated with (via cookies).
 [Authorize(AuthenticationSchemes = "Api"), IgnoreAntiforgeryToken, AllowAnonymous]
+[System.Diagnostics.CodeAnalysis.SuppressMessage(
+    "Major Code Smell",
+    "S6961:API Controllers should derive from ControllerBase instead of Controller",
+    Justification = "Seems to be a false positive as per line 62.")]
```
Comment (Piedone) [inline Lombiq.TrainingDemo/Controllers/ApiController.cs]:
```suggestion
    Justification = "Can't be changed due to line 62. Will be applicable after an Orchard upgrade due to " +
        "https://github.com/OrchardCMS/OrchardCore/issues/16186 being fixed.")]
```

File: Lombiq.TrainingDemo/Controllers/YourFirstOrchardCoreController.cs [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 namespace Lombiq.TrainingDemo.Controllers;
 
+[Route("[controller]/{action=Index}")]
```
Comment (Piedone) [inline Lombiq.TrainingDemo/Controllers/YourFirstOrchardCoreController.cs]: Neither of the actions can be accessed by their documented/configured URLs anymore.

File: Lombiq.TrainingDemo/Controllers/CrossTenantServicesController.cs [hunk 3]
```diff
@@ -24,6 +24,7 @@ namespace Lombiq.TrainingDemo.Controllers;
 // This is a controller just for the sake of easy demonstration, you can do the same thing anywhere. In the Index
 // action, we'll fetch content items from another tenant with the IContentManager service that you already know. This is
 // just an example though, really you can access any other service as well.
+[Route("[controller]/{action=Index}")]
```
Comment (Piedone) [inline Lombiq.TrainingDemo/Controllers/CrossTenantServicesController.cs]: /CrossTenantServices?contentItemId=ID doesn't work anymore.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: Lombiq.TrainingDemo/Controllers/CrossTenantServicesController.cs (hunk 3 above)
Comment (Psichorex) [reply Lombiq.TrainingDemo/Controllers/CrossTenantServicesController.cs]: Do we actually want this rule? The rule is generally saying that if there is an action level `[Route]` then there should be a Controller level `[Route]` aswell. So everything should be accessible via ControllerRoute/ActionRoute. We didn't really utilize this at all and I don't see the point why would we want for example an `api/somethings/paramteres` route to now be `ControllerRoute/ActionRoute/api/somethings/parameters`.
What do you think?

Comment (Piedone) [reply Lombiq.TrainingDemo/Controllers/CrossTenantServicesController.cs]: Yeah, let's not use this, then.

**Review by Piedone — APPROVED** [review]

**Pull request merged**